unblob/handlers/archive/_safe_tarfile.py

```diff
@@ -0,0 +1,101 @@
+import os
+from pathlib import Path
+from tarfile import SUPPORTED_TYPES, ExtractError, TarFile
+
+from structlog import get_logger
+
+from unblob.extractor import is_safe_path
+
+logger = get_logger()
+
+
+class SafeTarFile(TarFile):
+    def extract(  # noqa: C901
+        self, member, path="", set_attrs=True, *, numeric_owner=False
+    ):
```

Comment:
The changes in this function can be implemented with minor copy-paste if you reuse the original method via `super().extract(...)`.